### PR TITLE
chore: validate git installation upfront

### LINKED ISSUE #154: Validate whether if Git is installed in advance
### Motivation
The utility assumes `git` is already installed and fails miserably if that is not the case (throws an error). 

### Proposed Solution
IMO, it is good to validate the respective case upfront and warn appropriately.

### Alternatives
N/A

### Additional Context
N/A

### PR DESCRIPTION
**Pull request by jamesgeorge007**
closes #154

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
\cc @sendilkumarn

File: .bin/create-rust-webpack.js [hunk 1]
```diff
@@ -18,6 +29,11 @@ function run(cmd, args, opts) {
   }
 }
 
+if (!validateInstallation("git help -g")) {
```
Comment (sendilkumarn) [inline .bin/create-rust-webpack.js]: Can we rename the function too 👍 

```suggestion
if (!IsPresent("git --version")) {
```

File: .bin/create-rust-webpack.js [hunk 2]
```diff
@@ -18,6 +29,11 @@ function run(cmd, args, opts) {
...
+  console.log('\n git is required to be installed!');
```
Comment (sendilkumarn) [inline .bin/create-rust-webpack.js]:
```suggestion
  console.log('\n git is required!');
```

File: .bin/create-rust-webpack.js [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
 
-const { spawnSync } = require("child_process");
+const { execSync, spawnSync } = require("child_process");
 const fs = require("fs");
 const path = require("path");
 const cpr = require("cpr");
 const rimraf = require("rimraf");
 
+function validateInstallation(dep) {
+  let installationStatus;
```
Comment (sendilkumarn) [inline .bin/create-rust-webpack.js]:
```
try {
         execSync(dep, {stdio: 'ignore'});
         return true;
} catch (...) {
         return false;
}
```

**Comment by sendilkumarn** [comment]
Thanks for the PR @jamesgeorge007

**Comment by jamesgeorge007** [comment]
\cc @sendilkumarn

File: .bin/create-rust-webpack.js [hunk 4]
```diff
@@ -1,11 +1,21 @@
...
 const path = require("path");
 const cpr = require("cpr");
 const rimraf = require("rimraf");
 
+function isPresent(dep) {
+  let installationStatus;
+  try {
+    execSync(dep, {stdio: 'ignore'});
+    return true;
+  } catch (...) {
```
Comment (sendilkumarn) [inline .bin/create-rust-webpack.js]: I meant `...` to replace the `error`.

File: .bin/create-rust-webpack.js [hunk 5]
```diff
@@ -18,6 +28,11 @@ function run(cmd, args, opts) {
   }
 }
 
+if (!isPresent("git help -g")) {
```
Comment (sendilkumarn) [inline .bin/create-rust-webpack.js]: use `git --version` command here

**Comment by jamesgeorge007** [comment]
\cc @sendilkumarn

File: .bin/create-rust-webpack.js [hunk 6]
```diff
@@ -1,11 +1,21 @@
...
 
-const { spawnSync } = require("child_process");
+const { execSync, spawnSync } = require("child_process");
 const fs = require("fs");
 const path = require("path");
 const cpr = require("cpr");
 const rimraf = require("rimraf");
 
+function isPresent(dep) {
+  let installationStatus;
```
Comment (sendilkumarn) [inline .bin/create-rust-webpack.js]: Now can we remove this variable?

Comment (jamesgeorge007) [reply .bin/create-rust-webpack.js]: oops!

**Comment by jamesgeorge007** [comment]
\cc @sendilkumarn

**Comment by sendilkumarn** [comment]
oops I merged too fast, now I rebased and squash-merged.

**Pull request merged**